### PR TITLE
Support multiple video profiles for onchain jobs

### PR DESCRIPTION
**Pull request by ericxtang**
Now we pass a comma-separated video profile list onto the blockchain.  The transcoder will support multiple profiles.

We also got rid of the EOF field in HLSSegment.  We already have a way to end the stream by sending a FinishMsg.  Having multiple ways to finish a stream is causing some confusion.

We also removed a lot of unused old code in LPMS.

### CODE REVIEW

File: cmd/livepeer/livepeer.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func main() {
 	bootnode := flag.Bool("bootnode", false, "Set to true if starting bootstrap node")
 	transcoder := flag.Bool("transcoder", false, "Set to true to be a transcoder")
 	maxPricePerSegment := flag.Int("maxPricePerSegment", 1, "Max price per segment for a broadcast job")
-	transcodingOptions := flag.String("transcodingOptions", "P360p30fps16x9", "Transcoding options for broadcast job")
+	transcodingOptions := flag.String("transcodingOptions", "P240p30fps16x9,P360p30fps16x9", "Transcoding options for broadcast job")
```
Comment (dob) [inline cmd/livepeer/livepeer.go:88]: does this mean that by default we'll request transcoding to two profiles?

File: server/mediaserver.go [hunk 2]
```diff
@@ -72,8 +72,15 @@ func NewLivepeerServer(rtmpPort string, httpPort string, ffmpegPath string, lpNo
 func (s *LivepeerServer) StartMediaServer(ctx context.Context, maxPricePerSegment int, transcodingOptions string) error {
 	if s.LivepeerNode.Eth != nil {
 		BroadcastPrice = big.NewInt(int64(maxPricePerSegment))
-		BroadcastJobVideoProfile = types.VideoProfileLookup[transcodingOptions]
-		glog.Infof("Transcode Job Price: %v, Transcode Job Type: %v", BroadcastPrice, BroadcastJobVideoProfile)
+		bProfiles := make([]types.VideoProfile, 0)
+		for _, opt := range strings.Split(transcodingOptions, ",") {
```
Comment (dob) [inline server/mediaserver.go:76]: Should we also trim whitespace, in case the opts were passed in as comma separated with a space?

**Review by dob — APPROVED** [review]
Looks good to me :ship:

File: cmd/livepeer/livepeer.go (hunk 1 above)
Comment (ericxtang) [reply cmd/livepeer/livepeer.go:88]: Yes

File: server/mediaserver.go (hunk 2 above)
Comment (ericxtang) [reply server/mediaserver.go:76]: Good point

**Pull request merged**